packages/webvh/src/dids/WebvhDidResolver.ts

```diff
@@ -0,0 +1,128 @@
+import type { AgentContext, DidResolutionResult, DidResolver } from '@credo-ts/core'
+
+import { DidDocument } from '@credo-ts/core'
+import { resolveDID } from 'didwebvh-ts'
+
+import { DIDWebvhCrypto } from './didWebvhUtil'
+
+export class WebvhDidResolver implements DidResolver {
+  public readonly supportedMethods = ['webvh']
+  public readonly allowsCaching = false
+  public readonly allowsLocalDidRecord = false
+
+  public async resolve(agentContext: AgentContext, did: string): Promise<DidResolutionResult> {
+    const didDocumentMetadata = {}
+
+    try {
+      return await this.resolveDidDoc(agentContext, did)
+    } catch (error) {
+      agentContext.config.logger.error(`Error resolving DID: ${error}`)
+      return {
+        didDocument: null,
+        didDocumentMetadata,
+        didResolutionMetadata: {
+          error: 'notFound',
+          message: `resolver_error: Unable to resolve did '${did}': ${error}`,
+        },
+      }
+    }
+  }
+
+  public getBaseUrl = (id: string) => {
+    // Handle DID URL paths - split on first slash to separate DID from URL path
+    const slashIndex = id.indexOf('/')
+    let didPart = id
+    let urlPath = ''
+    
+    if (slashIndex !== -1) {
+      didPart = id.substring(0, slashIndex)
+      urlPath = id.substring(slashIndex)
+    }
+    
+    const parts = didPart.split(':');
+    if (!didPart.startsWith('did:webvh:') || parts.length < 4) {
+      throw new Error(`${id} is not a valid did:webvh identifier`);
+    }
+  
+    // Extract domain and path parts from the DID (after did:webvh:CID)
+    const domainAndPathParts = parts.slice(3)
+    const protocol = domainAndPathParts.join(':').includes('localhost') ? 'http' : 'https';
+  
+    // First part is domain, rest are path components
+    const domain = domainAndPathParts[0]
+    const pathComponents = domainAndPathParts.slice(1)
+    
+    // Check if domain has port
+    const [host, port] = domain.split(':')
+    const normalizedHost = port ? `${host}:${port}` : host;
+    
+    // Build the base URL
+    let baseUrl = `${protocol}://${normalizedHost}`
+    if (pathComponents.length > 0) {
+      baseUrl += '/' + pathComponents.join('/')
+    }
+    
+    // Append the URL path if present
+    if (urlPath) {
+      baseUrl += urlPath
+    }
+  
+    return baseUrl;
+  }
+
+  public async resolveResource(agentContext: AgentContext, resourceUrl: string) {
+    try {
+      agentContext.config.logger.debug(`Attempting to resolve resource: ${resourceUrl}`)
+      
+      // Use the getBaseUrl method to parse the did:webvh resource URL
+      const httpsUrl = this.getBaseUrl(resourceUrl)
+      
+      agentContext.config.logger.debug(`Fetching resource from: ${httpsUrl}`)
+      
+      // Fetch the resource using native fetch
+      const response = await fetch(httpsUrl)
```

Comment:
Should use fetch from the agent config. `agentContext.config.agentDependencies.fetch`